### PR TITLE
chore(setup): update Husky usage

### PR DESCRIPTION
**Pull request by merrywhether**
## Proposed Changes

Found while working on #197, pulled out to not force coupling.

- Updates to using [new Husky v9 commands](https://github.com/typicode/husky/releases/tag/v9.0.1)
Deprecated `install` and `add` commands were causing lifecycle events to fail. Verified that hooks are still installed and functional after these changes.

- Adds `packageManager` entry to `package.json` to enable [Corepack](https://nodejs.org/api/corepack.html) compatibility
References same version as referenced in setup/env. For Corepack users this will prompt automatic installation of the correct version, and will be noop for everyone else.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,14 +13,14 @@
...
     "commit": "git-cz",
     "commit-msg": "commitlint --edit $1",
-    "husky-setup": "path-exists .husky/commit-msg || (husky install && pnpm husky-setup:commit-msg && pnpm husky-setup:post-merge && pnpm husky-setup:pre-commit)",
-    "husky-setup:commit-msg": "npx husky add .husky/commit-msg 'pnpm run commit-msg'",
-    "husky-setup:post-merge": "npx husky add .husky/post-merge 'pnpm run setup'",
-    "husky-setup:pre-commit": "npx husky add .husky/pre-commit 'pnpm run pre-commit'",
+    "husky-setup": "path-exists .husky/commit-msg || (husky init && pnpm husky-setup:commit-msg && pnpm husky-setup:post-merge && pnpm husky-setup:pre-commit)",
+    "husky-setup:commit-msg": "echo 'pnpm run commit-msg' > .husky/commit-msg",
+    "husky-setup:post-merge": "echo 'pnpm run setup' > .husky/post-merge",
+    "husky-setup:pre-commit": "echo 'pnpm run pre-commit' > .husky/pre-commit",
```
Comment (merrywhether) [inline package.json:19]: Changes per [Husky v9 changelog](https://github.com/typicode/husky/releases/tag/v9.0.1)

File: package.json [hunk 2]
```diff
@@ -13,14 +13,14 @@
...
     "lint": "eslint index.js --fix",
     "lint:ci": "eslint index.js",
     "pre-commit": "pnpm lint && pnpm test",
-    "prepare": "is-ci || pnpm husky-setup",
+    "prepare": "husky",
```
Comment (merrywhether) [inline package.json:23]: Automatically updated by `husky init`

File: package.json [hunk 3]
```diff
@@ -101,5 +101,6 @@
     "github": {
       "release": true
     }
-  }
+  },
+  "packageManager": "pnpm@8.1.1"
```
Comment (yowainwright) [inline package.json:105]: Initially, I have gone with `package-manager-strict=false` in the `.npmrc` file vs this change. What are your thoughts?

**Review by yowainwright — APPROVED** [review]
LGTM! Besides the `corepack` update. Looks like that causes CI to fail. What are thoughts?

**Comment by merrywhether** [comment]
I've removed the corepack config addition from this branch in order to make the Husky changes easily mergable. I'm going to poke at corepack solutions a bit since I'm curious about the best options for supporting both types of users. Without the entry, anything that triggers `pnpm` has this immediate side effect:

```
! The local project doesn't define a 'packageManager' field. Corepack will now add one referencing pnpm@8.15.7+sha256.50783dd0fa303852de2dd1557cd4b9f07cb5b018154a6e76d0f40635d6cee019.
```
(picking the current default version in my default mise node version apparently)

We're in a weird limbo were people are starting to enable corepack as the default on their systems, but Node hasn't made it on by default quite yet more broadly.

**Pull request merged**